src/toxcore/onion/client/nodes_pool.rs

```diff
@@ -52,6 +56,47 @@ impl NodesPool {
     pub fn len(&self) -> usize {
         self.nodes.len()
     }
+
+    /// Build new random onion path with first UDP node.
+    pub fn udp_path(&self) -> Option<OnionPath> {
+        if self.len() < MIN_NODES_POOL_SIZE {
+            return None;
+        }
+        // non-empty nodes pool will always return some node
+        let node_1 = self.rand().unwrap();
+        let mut node_2;
+        loop {
+            node_2 = self.rand().unwrap();
+            if node_2 != node_1 {
+                break;
+            }
+        }
+        let mut node_3;
+        loop {
+            node_3 = self.rand().unwrap();
+            if node_3 != node_1 && node_3 != node_2 {
```

Comment:
What if there is less than 3 nodes in the pool? Will you enter an infinite loop?